asmcomp/arm64/emit.mlp

```diff
@@ -742,43 +754,47 @@ let emit_instr env i =
         assert (n mod 16 = 0);
         emit_stack_adjustment (-n);
         env.stack_offset <- env.stack_offset + n
-    | Lop(Iload(size, addr, _mut)) ->
+    | Lop(Iload { memory_chunk; addressing_mode; is_atomic }) ->
+        if is_atomic then ` dmb ishld\n`;
+        let load_type = if is_atomic then "ldar" else "ldr" in
```

Comment:
`Iload` can now emit `ldars`, `ldarsb`, `ldarsh` and `ldarsw`. Are we sure that these instructions are available? In the [Armv8 ISA](https://developer.arm.com/documentation/100076/0100/a64-instruction-set-reference/a64-data-transfer-instructions/ldar), I only see `ldar`, `ldarb` and `ldarh` available. And clang also seems to not generate these instructions: https://godbolt.org/z/aveq434v9.

Reply:
Atomic.get will generate a word-sized atomic load (of the "val" kind) I think, so other sizes are not needed.

Reply:
Fixed with merge of https://github.com/ctk21/ocaml/pull/1